src/Http/Action/Security/SsoRedirectAction.php

```diff
@@ -0,0 +1,158 @@
+     * @param string                                  $clientSecret
+     * @param string                                  $redirectUri
+     * @param string                                  $resourceUri
+     * @param string                                  $tokenUrl
+     * @param Client                                  $httpClient
+     */
+    public function __construct(
+        Sentinel $sentinel,
+        AccountManagement $accounts,
+        Routing\Generator\UrlGeneratorInterface $urlGenerator,
+        int $clientId,
+        string $clientSecret,
+        string $redirectUri,
+        string $resourceUri,
+        string $tokenUrl,
+        Client $httpClient
+    ) {
+        $this->sentinel     = $sentinel;
+        $this->accounts     = $accounts;
+        $this->urlGenerator = $urlGenerator;
+        $this->clientId     = $clientId;
+        $this->clientSecret = $clientSecret;
+        $this->redirectUri  = $redirectUri;
+        $this->resourceUri  = $resourceUri;
+        $this->tokenUrl     = $tokenUrl;
+        $this->httpClient   = $httpClient;
+    }
+
+    /**
+     * @param HttpFoundation\Request $request
+     *
+     * @throws \OpenCFP\Infrastructure\Auth\UserExistsException
+     * @throws \Symfony\Component\Routing\Exception\RouteNotFoundException
+     * @throws \Symfony\Component\Routing\Exception\MissingMandatoryParametersException
+     * @throws \Symfony\Component\Routing\Exception\InvalidParameterException
+     * @throws \OpenCFP\Infrastructure\Auth\UserNotFoundException
+     * @throws \InvalidArgumentException
+     *
+     * @return HttpFoundation\Response
+     */
+    public function __invoke(HttpFoundation\Request $request): HttpFoundation\Response
+    {
+        try {
+            $response = $this->httpClient->post($this->tokenUrl, [
+                'form_params' => [
+                    'grant_type'    => 'authorization_code',
+                    'client_id'     => $this->clientId,
+                    'client_secret' => $this->clientSecret,
+                    'redirect_uri'  => $this->redirectUri,
+                    'code'          => $request->get('code'),
+                ],
+            ]);
+        } catch (\Exception $e) {
+            $request->getSession()->set('flash', [
+                'type'  => 'error',
+                'short' => 'Error',
+                'ext'   => 'We were unable to authenticate with OpenCFP Central. Please try again',
+            ]);
+            $url = $this->urlGenerator->generate('login');
+
+            return new HttpFoundation\RedirectResponse($url);
+        }
+
+        /**
+         * What do we do with the access token?
+         * Check if the user exists
+         * -> if they don't, generate a random password and then create and activate an account
+         * manually log them in
+         * redirect them to their dashboard
+         */
+        $details = \json_decode((string) $response->getBody(), true);
```

Comment:
Should we check here whether the response body actually contained valid JSON, before moving on and assuming it does?